### PR TITLE
Move certs into ssl folder.

### PR DESCRIPTION
**Pull request by mrstegeman**
Resurrecting #143.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,9 +119,9 @@ $ yarn
  The HTTPS server looks for privatekey.pem and certificate.pem. You can use a real certificate or generate a self-signed one by following the steps below.
 
  ```
- $ openssl genrsa -out privatekey.pem 2048
- $ openssl req -new -sha256 -key privatekey.pem -out csr.pem
- $ openssl x509 -req -in csr.pem -signkey privatekey.pem -out certificate.pem
+ $ openssl genrsa -out ssl/privatekey.pem 2048
```
Comment (hobinjk) [inline README.md]: Potentially prefix this with `mkdir ssl`

File: src/app.js [hunk 2]
```diff
@@ -53,6 +53,19 @@ if (fs.existsSync('notunnel')) {
   });
 }
 
+// Move certificates, if necessary. If this throws an error, let it bubble up.
+if (fs.existsSync('privatekey.pem')) {
```
Comment (hobinjk) [inline src/app.js]: These need the `ssl` directory to already exist

File: tools/deploy-certificates.sh [hunk 3]
```diff
@@ -6,9 +6,9 @@ gateway_dir="/home/pi/mozilla-iot/gateway"
 umask 077
 
 # Copy in the new certs.
-cp "$RENEWED_LINEAGE/chain.pem" "$gateway_dir/chain.pem"
-cp "$RENEWED_LINEAGE/cert.pem" "$gateway_dir/certificate.pem"
-cp "$RENEWED_LINEAGE/privkey.pem" "$gateway_dir/privatekey.pem"
+cp "$RENEWED_LINEAGE/chain.pem" "$gateway_dir/ssl/chain.pem"
```
Comment (hobinjk) [inline tools/deploy-certificates.sh]: This will likely also need a mkdir ssl based on whether ssl exists

File: tools/make-self-signed-cert.sh [hunk 4]
```diff
@@ -2,6 +2,6 @@
 
 set -x
 
-openssl genrsa -out privatekey.pem 2048
-openssl req -new -sha256 -key privatekey.pem -out csr.pem -subj '/CN=www.toizom.com/O=MozillaIoT Gateway/C=US'
-openssl x509 -req -in csr.pem -signkey privatekey.pem -out certificate.pem
+openssl genrsa -out ssl/privatekey.pem 2048
```
Comment (hobinjk) [inline tools/make-self-signed-cert.sh]: `mkdir -p ssl` before

File: tools/manual-renew-certs.sh [hunk 5]
```diff
@@ -121,11 +121,11 @@ rm -f "${pagekite_pidfile}"
 
 echo "Copying in new certificates."
 cp "${moziot_dir}/etc/live/${domain}/cert.pem" \
```
Comment (hobinjk) [inline tools/manual-renew-certs.sh]: Potentially `mkdir -p ssl` before

**Comment by mrstegeman** [comment]
This is actually going to become rather important if we decide to build a docker image. Having the certs in a separate directory will allow for data persistence. We'll also need to move the settings database, but that can be done separately.

**Comment by dhylands** [comment]
Presumably then the database and log files should go into separate directories as well?

We should file a bug to handle this in regards to updates as well. i.e. An update will need to move the certs from their current directory to the new directory.

**Comment by mrstegeman** [comment]
@dhylands this PR handles moving the certs. And yes, logs and such should go elsewhere.

**Pull request merged**